src/olympia/devhub/views.py

```diff
@@ -2074,3 +2079,99 @@ def logout(request):
     logout_user(request, response)
 
     return response
+
+
+VERIFY_EMAIL_STATE = {
+    'email_verified': 'email_verified',
+    'email_suppressed': 'email_suppressed',
+    'verification_expired': 'verification_expired',
+    'verification_pending': 'verification_pending',
+    'verification_failed': 'verification_failed',
+    'verification_timedout': 'verification_timedout',
+    'confirmation_pending': 'confirmation_pending',
+    'confirmation_unauthorized': 'confirmation_unauthorized',
+}
+
+RENDER_BUTTON_STATES = [
+    VERIFY_EMAIL_STATE['email_suppressed'],
+    VERIFY_EMAIL_STATE['verification_expired'],
+    VERIFY_EMAIL_STATE['verification_failed'],
+    VERIFY_EMAIL_STATE['verification_timedout'],
+    VERIFY_EMAIL_STATE['confirmation_unauthorized'],
+]
+
+
+def get_button_text(state):
+    if state == VERIFY_EMAIL_STATE['email_suppressed']:
+        return gettext('Verify email')
+
+    return gettext('Try again')
+
+
+@login_required
+def email_verification(request):
+    data = {'state': None}
+    email_verification = request.user.email_verification
+    suppressed_email = request.user.suppressed_email
+
+    if request.method == 'GET':
+        if email_verification:
+            if not email_verification.is_expired:
+                if email_verification.status_pending:
+                    if email_verification.is_timedout:
+                        data['state'] = VERIFY_EMAIL_STATE['verification_timedout']
+                    else:
+                        data['state'] = VERIFY_EMAIL_STATE['verification_pending']
+                elif email_verification.status_failed:
+                    data['state'] = VERIFY_EMAIL_STATE['verification_failed']
+                elif email_verification.status_delivered:
+                    code = request.GET.get('code')
+
+                    if code:
```

Comment:
```suggestion
                    if code := request.GET.get('code'):
```

Reply:
Didn't know I could make it even more compact!